<?xml version="1.0"?>

<launch>

    <let name="base_serial_port"        value="/dev/ttyACM0"/>
    <arg name="is_serial_port"          default="/dev/ttyUSB0"/>
    <let name="base_host"               value="192.168.1.122"/>
    <let name="base_port1"              value="16140"/>
    <let name="base_port2"              value="16145"/>
    <let name="compass_serial_port"     value="/dev/ttyACM1"/>
    <let name="rover_host1"             value="192.168.1.108"/>
    <let name="rover_port1"             value="16140"/>
    <let name="rover_host2"             value="192.168.1.9"/>
    <let name="rover_port2"             value="16140"/>
    <let name="base_type"               value="moving"/>
    <let name="rover_serial_port1"      value="/dev/ttyACM0"/>
    <let name="rover_serial_port2"      value="/dev/ttyACM0"/>


    <node pkg="ublox_read_2" exec="ublox_ros" name="base">
        <param name="serial_port"       value="$(var base_serial_port)"/>
        <param name="local_host"        value="$(var base_host)"/>
        <param name="local_port"        value="$(var base_port1)"/>
        <param name="rover_host"        value="$(var rover_host1)"/>
        <param name="rover_port"        value="$(var rover_port1)"/>
        <param name="local_host2"       value="$(var base_host)"/>
        <param name="local_port2"       value="$(var base_port2)"/>
        <param name="rover_host2"       value="$(var rover_host2)"/>
        <param name="rover_port2"       value="$(var rover_port2)"/>

        <param from="$(find-pkg-share ublox_read_2)/params/ublox.yaml"/>
        <param name="rover_quantity"    value="2"/>
        <param name="base_type"         value="$(var base_type)" />
    </node>

    <node name="rover1" pkg="ublox_read_2" exec="ublox_ros">
        <param name="serial_port"       value="$(var rover_serial_port1)"/>
        <param name="local_host"        value="$(var rover_host1)"/>
        <param name="local_port"        value="$(var rover_port1)"/>
        <param name="base_host"         value="$(var base_host)"/>
        <param name="base_port"         value="$(var base_port1)"/>

        <param from="$(find-pkg-share ublox_read_2)/params/ublox.yaml"/>
        <param name="rover_quantity"    value="0"/>
        <param name="base_type"         value="$(var base_type)" />
        <param name="message_rate"      value="5"/>
    </node>

    <node name="rover2" pkg="ublox_read_2" exec="ublox_ros">
        <param name="serial_port"       value="$(var rover_serial_port2)"/>
        <param name="local_host"        value="$(var rover_host2)"/>
        <param name="local_port"        value="$(var rover_port2)"/>
        <param name="base_host"         value="$(var base_host)"/>
        <param name="base_port"         value="$(var base_port2)"/>

        <param from="$(find-pkg-share ublox_read_2)/params/ublox.yaml"/>
        <param name="rover_quantity"    value="0"/>
        <param name="base_type"         value="$(var base_type)" />
        <param name="message_rate"      value="5"/>
    </node>
</launch>
